File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int mini=prices[0];
        int maxi=0;
        for(int i=1;i<n;i++){
            maxi=max(maxi,prices[i]-mini);
            mini=min(mini,prices[i]);
        }
        return max(maxi,prices[n-1]-mini);
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini=100000;
        int ans=0;
        int n=prices.length;
        for(int i=0;i<n;i++){
            mini=Math.min(mini,prices[i]);
            ans=Math.max(prices[i]-mini,ans);
        }
        return ans;
    }
}